exporter/kinesisexporter/producer.go

```diff
@@ -0,0 +1,78 @@
+package kinesisexporter
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	omnition "github.com/signalfx/omnition-kinesis-producer"
+	"github.com/signalfx/omnition-kinesis-producer/loggers/kpzap"
+	"go.uber.org/zap"
+	"time"
+)
+
```

Comment:
Could a comment be added here for what the interface is and why we have this interface? i.e. for testability